docs/core/testing/unit-testing-mstest-migration-from-v1-to-v3.md

```diff
@@ -0,0 +1,290 @@
+
+- **Increased Code Reliability with New Analyzers**: MSTest v3 built-in code analyzers enforce best practices, helping to catch issues early and promoting cleaner, more maintainable test code:
+  - **Proactive Issue Detection**: The analyzers provide real-time feedback, suggesting improvements and flagging potential issues as you write your tests.
+  - **Stronger Type-Safe Assertions**: Replacing ambiguous overloads, MSTest v3 enforces type-safe assertions, reducing the risk of incorrect test behavior and making tests more reliable.
+  - **Cleaner Code, Less Maintenance**: By aligning your tests with MSTest standards and best practices, MSTest v3 helps you maintain a cleaner, more manageable codebase, reducing technical debt over time.
+
+- **Greater Flexibility and Extensibility**: MSTest v3 supports advanced testing scenarios, including dynamic data sources and in-assembly parallel execution. This flexibility enables more sophisticated testing approaches and speeds up test suites without complex configurations.
+
+By upgrading, you’re setting up your tests to be faster, more reliable, and adaptable to future .NET developments, positioning your project for long-term success and easier maintenance.
+
+---
+
+## Steps to Migrate
+
+### 1. Remove Assembly Reference
+
+For projects using MSTest v1 through assembly references, it’s common to find references to either or both of the following DLLs:
+
+- `Microsoft.VisualStudio.QualityTools.UnitTestFramework`
+- `Microsoft.VisualStudio.TestPlatform.TestFramework`
+
+In non-SDK style projects, these references are often added through Visual Studio rather than by directly editing the XML. To remove these references using the Visual Studio GUI:
+
+1. **Open Solution Explorer** in Visual Studio.
+2. **Right-click on the project** using MSTest and select **Properties**.
+3. Navigate to the **References** tab.
+4. Locate and **select the MSTest DLL references** (`Microsoft.VisualStudio.QualityTools.UnitTestFramework` and/or `Microsoft.VisualStudio.TestPlatform.TestFramework`).
+5. **Right-click** the selected reference(s) and choose **Remove**.
+
+For those comfortable editing XML directly, you can manually remove the references from your `.csproj` file as follows:
+
+```xml
+<Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+    <Private>False</Private>
+</Reference>
+```
+
+### 2. Update Your Project
+
+You can update your project to MSTest v3 in one of two ways:
+
+- **Install via NuGet Package**: Install the latest [MSTest](https://www.nuget.org/packages/MSTest) package using the NuGet Package Manager in Visual Studio or by running the following command in the NuGet Package Manager Console:
+
+    ```shell
+    Install-Package MSTest.TestFramework -Version 3.3.1
+    ```
+
+- **Or update the project file directly**: Update your `.csproj` file to specify the MSTest SDK version.
```

Comment:
same feedback as before. This is probably largely incompatible with non-sdk projects, or with the setup they currently have in a sdk-style project. It would be more usable (in my eyes) to first show the direct package reference.

Reply:
She is first mentioning the metapackage so it's probably good. Am I missing your point?

@engyebrahim It's worth mentioning explicitly this is only compatible for SDK style projects.

Reply:
I think you do miss my point. :)

The projects you are trying to upgrade are likely non-sdk style csproj and are using dlls or are using MSTest.TestFramework and MSTest.TestAdapter packages.

The proposed steps are for modern sdk-style projects, that already use MSTest nuget. 

Instead the steps should be aimed at people who want to do the least amount of work to get to mstestv3, and maybe they will eventually migrate to sdk style csproj, and maybe even MSTest SDK. 

So:

Remove dll  refs,
Install or update mstest.testframework and mstest.testadapter with the latest version
Fix code

After that:

Consider moving to sdk style csproj
consider moving to MSTest meta package
Consider movgin to mstest.sdk